### ISSUE TITLE
`Move to where clause` assist does not work for lifetime targets

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#19681]
```rust
pub fn from_slice<'input:$0 'facet, 'facet, T>(
    json: &'input [u8],
) -> Result<T, JsonError<'_>> where T: Facet<'facet> {}
```
should trigger it here
